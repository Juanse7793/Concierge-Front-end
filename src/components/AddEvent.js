import React, { useState } from 'react';
import TextField from '@mui/material/TextField';
// import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
// import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
// import { DatePicker } from '@mui/x-date-pickers/DatePicker';
import { useDispatch } from 'react-redux';
import { addEvent } from '../redux/reducers/events';

function AddEvent() {
  // eslint-disable-next-line camelcase
  const [start_date, setStart] = useState();
  // eslint-disable-next-line camelcase
  const [end_date, setEnd] = useState();
  const [name, setName] = useState();
  const [location, setLocation] = useState();
  const [price, setPrice] = useState();

  const dispatch = useDispatch();

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(addEvent({
      start_date, end_date, name, location, price,
    }));
    setName('');
    setLocation('');
    setPrice('');
    setStart('');
    setEnd('');
  };

  const sx = {
    width: '500px',
    color: 'black',
  };

  return (
    <div className="add-event-main">
      <div className="add-event-header">
        <h1 className="add-event-title">Add Event</h1>
      </div>
      <form className="add-event-form" onSubmit={handleSubmit}>
        <TextField
          value={name}
          onChange={(e) => setName(e.target.value)}
          sx={sx}
          id="event-name-input"
          label="Event Name"
          margin="normal"
        />
        <TextField
          value={location}
          onChange={(e) => setLocation(e.target.value)}
          sx={sx}
          id="event-location-input"
          label="Event Location"
          margin="normal"
        />
        <TextField
          value={price}
          onChange={(e) => setPrice(e.target.value)}
          sx={sx}
          id="event-price-input"
          label="Event Price"
          margin="normal"
        />
        <div className="add-event-dates">
          <h2 className="add-event-dates-title">Event dates</h2>
        </div>
        <TextField
          // eslint-disable-next-line camelcase
          value={start_date}
          onChange={(e) => setStart(e.target.value)}
          sx={sx}
          id="event-start-input"
          label="Event Start"
          margin="normal"
        />
        <TextField
          // eslint-disable-next-line camelcase
          value={end_date}
          onChange={(e) => setEnd(e.target.value)}
          sx={sx}
          id="event-end-input"
          label="Event End"
          margin="normal"
        />
        <div className="add-event-images">
          <h2 className="add-event-images-title">Event images</h2>
        </div>
        <input type="file" multiple id="new-event-image" />
        <input type="submit" id="submit-event" value="Add Event" />
      </form>
    </div>
  );
}

export default AddEvent;
