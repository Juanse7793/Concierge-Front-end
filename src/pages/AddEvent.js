import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { addEvent } from '../redux/reducers/events';
import { InputText, DateRange } from '../components/Inputs';

function AddEvent() {
  const [formData, setFormData] = useState({
    name: '',
    location: '',
    price: '',
    images: [],
    start_date: '',
    end_date: '',
  });

  const [dateRange, setDateRange] = useState([null, null]);
  const [startDate, endDate] = dateRange;

  const setInputData = (e) => {
    setFormData({ ...formData, [e.name]: e.value });
  };
  const fileSelectedHandler = (e) => {
    setFormData({ ...formData, images: [...e.target.files] });
  };

  const dispatch = useDispatch();
  const handleSubmit = (e) => {
    const data = new FormData();
    data.append('name', formData.name);
    data.append('location', formData.location);
    data.append('price', formData.price);
    data.append('start_date', startDate);
    data.append('end_date', endDate);
    formData.images.forEach((el, i) => {
      data.append('images[]', formData.images[i], el);
    });

    e.preventDefault();
    dispatch(addEvent(data));
  };

  return (
    <section className="add-event-main background column">
      <form className="add-event-form column" onSubmit={handleSubmit}>
        <h1 className="add-event-title black-glow">Add Event</h1>
        <InputText
          text="Name"
          value={formData.name}
          func={(e) => setInputData(e.target)}
        />
        <InputText
          text="Location"
          value={formData.location}
          func={(e) => setInputData(e.target)}
        />
        <InputText
          text="Price"
          value={formData.price}
          func={(e) => setInputData(e.target)}
        />
        <DateRange
          startDate={startDate}
          endDate={endDate}
          func={(e) => setDateRange(e)}
        />

        <h2 className="add-event-images-title black-glow">Event Images</h2>
        <input type="file" multiple id="new-event-image" name="images" onInput={fileSelectedHandler} accept="image/*" />
        <input type="submit" value="Add Event" className="pill white" />
      </form>
    </section>
  );
}

export default AddEvent;
