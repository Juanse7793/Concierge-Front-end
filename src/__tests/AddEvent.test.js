import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import AddEvent from '../pages/AddEvent';

const mockStore = configureMockStore();
const store = mockStore({ events: { adding: false } });

it('matches snapshot', () => {
  const component = renderer.create(
    <Provider store={store}>
      <AddEvent />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
