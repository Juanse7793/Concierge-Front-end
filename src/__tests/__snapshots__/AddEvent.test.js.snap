// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="add-event-main background column"
>
  <form
    className="add-event-form column"
    onSubmit={[Function]}
  >
    <h1
      className="add-event-title black-glow"
    >
      Add Event
    </h1>
    <div
      className="out "
    >
      <input
        autoComplete="off"
        className="pill green white-border"
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <label
        htmlFor="name"
      >
        Name
      </label>
    </div>
    <div
      className="out "
    >
      <input
        autoComplete="off"
        className="pill green white-border"
        id="location"
        name="location"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <label
        htmlFor="location"
      >
        Location
      </label>
    </div>
    <div
      className="out "
    >
      <input
        autoComplete="off"
        className="pill green white-border"
        id="price"
        name="price"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <label
        htmlFor="price"
      >
        Price
      </label>
    </div>
    <div
      className="out focus"
    >
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            className="pill green white-border"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <label>
        Dates
      </label>
    </div>
    <h2
      className="add-event-images-title black-glow"
    >
      Event Images
    </h2>
    <input
      accept="image/*"
      id="new-event-image"
      multiple={true}
      name="images"
      onInput={[Function]}
      type="file"
    />
    <input
      className="pill white"
      type="submit"
      value="Add Event"
    />
  </form>
</section>
`;
